src/main/java/hudson/plugins/mercurial/MercurialChangeSet.java

```diff
@@ -280,13 +280,14 @@ public void setParents(String parents) {
     private List<String> toList(String list) {
         list = list.trim();
         if(list.length()==0) return Collections.emptyList();
-        return Arrays.asList(list.split(" "));
+        return Arrays.asList(list.split("<file/>"));
     }
 
     /** |xmlescape handles a few cases that |escape does not */
     static final String CHANGELOG_TEMPLATE =
-            "<changeset node='{node}' author='{author|xmlescape}' rev='{rev}' date='{date}'>" +
```

Comment:
Please avoid gratuitous whitespace changes.

Reply:
Sorry, Eclipse auto-indented, and I hadn't noticed.